upstart/ua-premium.conf

```diff
@@ -0,0 +1,14 @@
+# ua-premium - Ubuntu Advantage Premium attach
+
+# This task is run on reboot to adapt to potential change in
+# kernels/environment across boot and update Ubuntu Advantage status
+# information.
+
```

Comment:
I am not asking for changes, but trying to understand why we try to attach literally on every boot. shouldn't we check to see if we are attached first?

Reply:
calls to `ua attach-premium` has a decorator that checks config.is_attached validate whether we're attached each call you make. it exits 0 in this case to indicate succesful end state (attached). I thought this was clearer than:
`ua status --format=json | grep -q '"attached": true'` || ua attach-premium

But maybe the above makes more sense and introduces less upstart log noise.

Reply:
Hrm, also I don't think I can express compound exec commands in upstart. I certainly can't in systemd.   If we surfaced a command that would exit 1 on unattached we could use something like
ExecCondtion=ua is-attached
ExecStart=ua attach-premium

Reply:
Additionally we could add a param --silent to attach-premium to avoid adding log messages about machine already attached (because we do noop if we are already attached)